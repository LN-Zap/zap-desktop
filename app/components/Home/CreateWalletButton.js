import React from 'react'
import { Flex } from 'rebass'
import { Button, Text } from 'components/UI'
import PlusCircle from 'components/Icon/PlusCircle'

const CreateWalletButton = ({ ...rest }) => (
  <Button {...rest} variant="secondary">
    <Flex alignItem="center">
      <Text color="lightningOrange">
        <PlusCircle width="22px" height="22px" />
      </Text>
      <Text lineHeight="22px" ml={2}>
        Create new wallet
      </Text>
    </Flex>
  </Button>
)
export default CreateWalletButton
