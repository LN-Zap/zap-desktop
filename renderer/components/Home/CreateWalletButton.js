import React from 'react'
import PropTypes from 'prop-types'
import { FormattedMessage } from 'react-intl'
import { Flex } from 'rebass'
import { Button, Text } from 'components/UI'
import PlusCircle from 'components/Icon/PlusCircle'
import messages from './messages'

const CreateWalletButton = ({ history, ...rest }) => (
  <Button onClick={() => history.push('/onboarding')} size="small" variant="secondary" {...rest}>
    <Flex alignItem="center">
      <Text color="lightningOrange">
        <PlusCircle height="22px" width="22px" />
      </Text>
      <Text lineHeight="22px" ml={2}>
        <FormattedMessage {...messages.create_wallet_button_text} />
      </Text>
    </Flex>
  </Button>
)
CreateWalletButton.propTypes = {
  history: PropTypes.object.isRequired,
}

export default CreateWalletButton
