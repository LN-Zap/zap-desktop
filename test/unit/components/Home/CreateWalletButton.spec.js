import React from 'react'

import { IntlProvider } from 'react-intl'

import { renderWithTheme } from '@zap/test/unit/__helpers__/renderWithTheme'
import CreateWalletButton from 'components/Home/CreateWalletButton'

const history = { push: jest.fn() }

describe('component.CreateWalletButton', () => {
  it('should render correctly', () => {
    const tree = renderWithTheme(
      <IntlProvider locale="en">
        <CreateWalletButton history={history} />
      </IntlProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
