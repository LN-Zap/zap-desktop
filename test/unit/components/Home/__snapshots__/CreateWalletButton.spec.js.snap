// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.CreateWalletButton should render correctly 1`] = `
<Button
  active={false}
  onClick={[Function]}
  processing={false}
  size="medium"
  variant="secondary"
>
  <Styled(styled.div)
    alignItem="center"
  >
    <Text
      color="lightningOrange"
    >
      <SvgPlusCircle
        height="22px"
        width="22px"
      />
    </Text>
    <Text
      lineHeight="22px"
      ml={2}
    >
      <FormattedMessage
        defaultMessage="Create new wallet"
        id="components.Home.create_wallet_button_text"
        values={Object {}}
      />
    </Text>
  </Styled(styled.div)>
</Button>
`;
